src/Handlers.jl

```diff
@@ -88,15 +126,15 @@ register!(r::Router, url, handler) = register!(r, "", url, handler)
 function register!(r::Router, method::String, url, handler)
     m = isempty(method) ? Any : typeof(Val(Symbol(method)))
     # get scheme, host, split path into strings & vals
-    uri = url isa String ? HTTP.URI(url) : url
+    uri = url isa String ? URI(url) : url
     s = uri.scheme
     sch = !isempty(s) ? typeof(get!(SCHEMES, s, Val(s))) : Any
     h = !isempty(uri.host) ? Val{Symbol(uri.host)} : Any
-    hand = handler isa Function ? HandlerFunction(handler) : handler
+    hand = handler isa Function ? RequestHandlerFunction(handler) : handler
```

Comment:
In general,
for both uri and handlers,
it might be nicer to have functions that 
convert things if they are not already the right type,
or do nothing if they are.
IIRC `convert` does this already
Then it could be written with dispatch.
And it is easier to control how and if different types are converted.
so one would just right `uri = convert(URI, url)`
and 
`hand = convert(Handler, handler)`
(This would require definging the `convert` for some of these)